pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -348,8 +352,27 @@ func (p *provider) getConfig(s v1alpha1.ProviderSpec) (*Config, *providerconfigt
 		return nil, nil, nil, fmt.Errorf("failed to get ebsVolumeEncrypted value: %v", err)
 	}
 	c.Tags = rawConfig.Tags
-	c.IsSpotInstance = rawConfig.IsSpotInstance
 	c.AssignPublicIP = rawConfig.AssignPublicIP
+	c.IsSpotInstance = rawConfig.IsSpotInstance
+	if c.IsSpotInstance != nil && *c.IsSpotInstance {
+		maxPrice, err := p.configVarResolver.GetConfigVarStringValueOrEnv(rawConfig.SpotInstanceConfig.MaxPrice, "SPOT_INSTANCE_MAX_PRICE")
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		c.SpotMaxPrice = pointer.StringPtr(maxPrice)
+
+		persistentRequest, err := p.configVarResolver.GetConfigVarBoolValueOrEnv(rawConfig.SpotInstanceConfig.PersistentRequest, "SPOT_INSTANCE_PERSISTENT_REQUEST")
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		c.SpotPersistentRequest = pointer.BoolPtr(persistentRequest)
+
+		interruptionBehavior, err := p.configVarResolver.GetConfigVarStringValueOrEnv(rawConfig.SpotInstanceConfig.InterruptionBehavior, "SPOT_INSTANCE_INTERRUPTION_BEHAVIOR")
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		c.SpotInterruptionBehavior = pointer.StringPtr(interruptionBehavior)
```

Comment:
Does it make sense to read this information from the environment? This can be useful to enforce spot instances, but also can be confusing. Also, different Machines might have different settings.

Reply:
Honestly I am indifferent about that so I can get rid of it ;)